annotationProcessor/src/main/java/org/corfudb/annotations/ObjectAnnotationProcessor.java

```diff
@@ -0,0 +1,642 @@
+                classElement.getEnclosedElements().stream()
+                .filter(x -> x.getAnnotation(Accessor.class) != null ||
+                        x.getAnnotation(MutatorAccessor.class) != null ||
+                        x.getAnnotation(Mutator.class) != null)
+                .count() > 0) {
+
+            try {
+                generateProxy(classElement);
+            } catch (IOException e) {
+                messager.printMessage(Diagnostic.Kind.ERROR, "Failed to process class "
+                        + classElement.getSimpleName() + " with IOException");
+            }
+        }
+    }
+
+    /** Return a SMR function name, extracting it from the annotations if available.
+     *
+     * @param smrMethod     The ExecutableElement to extract the name from.
+     * @return              An SMR Function name string.
+     */
+    public String getSMRFunctionName(ExecutableElement smrMethod) {
+        MutatorAccessor mutatorAccessor = smrMethod.getAnnotation(MutatorAccessor.class);
+        Mutator mutator = smrMethod.getAnnotation(Mutator.class);
+        return  (mutator != null && !mutator.name().equals("")) ? mutator.name() :
+                (mutatorAccessor != null && !mutatorAccessor.name().equals("")) ?
+                mutatorAccessor.name() : smrMethod.getSimpleName().toString();
+    }
+
+    /** Class to hold data about SMR Methods. */
+    class SMRMethodInfo {
+        ExecutableElement method;
+        TypeElement interfaceOverride;
+        boolean doNotAdd = false;
+
+        public SMRMethodInfo(ExecutableElement method, TypeElement interfaceOverride) {
+            this.method = method;
+            this.interfaceOverride = interfaceOverride;
+        }
+
+        public SMRMethodInfo(ExecutableElement method, TypeElement interfaceOverride, boolean doNotAdd) {
+            this.method = method;
+            this.interfaceOverride = interfaceOverride;
+            this.doNotAdd = doNotAdd;
+        }
+    }
+
+    /** Generate a proxy file.
+     *
+     * @param classElement  The element to generate a proxy file for
+     * @throws IOException  If we could not generate the proxy file
+     */
+    public void generateProxy(TypeElement classElement)
+            throws IOException
+    {
+        // Extract the package name for the class. We'll need this to generate the proxy file.
+        String packageName = elementUtils.getPackageOf(classElement).toString();
+        // Calculate the name of the proxy, which appends $CORFUSMR to the class name.
+        ClassName proxyName = ClassName.bestGuess(classElement.getSimpleName().toString()
+                + ICorfuSMR.CORFUSMR_SUFFIX);
+        // Also get the class name of the original class.
+        TypeName originalName = ParameterizedTypeName.get(classElement.asType());
+
+        // Generate the proxy class. The proxy class extends the original class and implements
+        // ICorfuSMR.
+        TypeSpec.Builder typeSpecBuilder = TypeSpec
```

Comment:
This method is way too long and hard to read and understand. Is it possible to break this up into smaller methods with more comments. Also a sample of the final output might help the the understanding even more.